eclair-core/src/main/scala/fr/acinq/eclair/payment/send/Recipient.scala

```diff
@@ -190,7 +190,7 @@ object BlindedRecipient {
  * Note that we don't need to support the case where we'd use multiple trampoline hops in the same route: since we have
  * access to the network graph, it's always more efficient to find a channel route to the last trampoline node.
  */
-case class ClearTrampolineRecipient(invoice: Bolt11Invoice,
+case class ClearTrampolineRecipient(invoice: Invoice,
```

Comment:
The name doesn't really make sense anymore, since this covers the following three distinct cases:

- recipient that supports trampoline without Bolt 12 (current Phoenix)
- recipient that doesn't support trampoline nor Bolt 12 (trampoline-to-legacy)
- recipient that doesn't support trampoline but supports Bolt 12 (trampoline-to-blinded-paths)

What about just renaming this `TrampolineRecipient` for now, and we'll see later if we actually want to split those cases in different `Recipient` instances?

Reply:
Done